src/wallet/storage.rs

```diff
@@ -157,3 +157,106 @@ impl FileData {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bip39::Mnemonic;
+    use bitcoin::Network;
+    use bitcoind::tempfile::tempdir;
+    // Generate a dummy mnemonic for testing
+
+    fn generate_dummy_mnemonic() -> String {
+        Mnemonic::generate(12).unwrap().to_string()
+    }
+
+    #[test]
+    fn test_wallet_store_init() {
+        let temp_dir = tempdir().unwrap();
+
+        let file_path = temp_dir.path().join("test_wallet.json");
+        let mnemonic = generate_dummy_mnemonic();
+
+        let wallet_store = WalletStore::init(
+            "test_wallet".to_string(),
+            &file_path,
+            Network::Bitcoin,
+            mnemonic,
+            "passphrase".to_string(),
+        )
+        .unwrap();
+
+        assert_eq!(wallet_store.wallet_name, "test_wallet");
+        assert_eq!(wallet_store.network, Network::Bitcoin);
+    }
+
+    #[test]
+    fn test_write_and_read_to_disk() {
+        let temp_dir = tempdir().unwrap();
+        let file_path = temp_dir.path().join("test_wallet.json");
+        let mnemonic = generate_dummy_mnemonic();
+
+        let original_wallet_store = WalletStore::init(
+            "test_wallet".to_string(),
+            &file_path,
+            Network::Bitcoin,
+            mnemonic.clone(),
+            "passphrase".to_string(),
+        )
+        .unwrap();
+
+        original_wallet_store.write_to_disk(&file_path).unwrap();
+
+        let read_wallet = WalletStore::read_from_disk(&file_path).unwrap();
+
+        assert_eq!(read_wallet.wallet_name, original_wallet_store.wallet_name);
+        assert_eq!(read_wallet.network, original_wallet_store.network);
+    }
```

Comment:
These two could be combined into one. read/write test already implies init works too. Keep it in a single test functions. and assert for both these cases. You would wanna assert for all the fields, that maximizes coverage ratio..

Reply:
Yes, make sense... While writing test for init methods I was aware of it but wanted your confirmation. Cool I will make changes.